pkg/util/log/log.go

```diff
@@ -396,6 +396,17 @@ func Info(v ...interface{}) {
 	}
 }
 
+// Infoc logs at the info level with context
+func Infoc(message string, context ...interface{}) {
+	if logger != nil && logger.inner != nil && logger.shouldLog(seelog.InfoLvl) {
+		logger.inner.SetContext(context)
```

Comment:
I don't think this is thread safe, which is probably a bit of a show-stopper. Adding a lock would have a performance penalty for sure, but given the fact we do stuff like log entry scrubbing it might be small enough we can live with it. Should definitely be benchmarked in a concurrent scenario though.

Reply:
Good catch! I'll have a closer look at a concurrent scenario 👍

Reply:
Confirmed this is not thread safe, started two routines logging with different contexts and these got mixed up a few times 😞  

Here are the results adding a `contextLock` in `Infoc`:
```
BenchmarkLogVanilla-4           	  443283	      2850 ns/op	    1118 B/op	      14 allocs/op
BenchmarkLogVanillaLevels-4     	16086051	        77.3 ns/op	      64 B/op	       2 allocs/op
BenchmarkLogScrubbing-4         	   62612	     18072 ns/op	    6576 B/op	      38 allocs/op
BenchmarkLogScrubbingLevels-4   	32663391	        39.0 ns/op	      16 B/op	       1 allocs/op
BenchmarkLogScrubbingMulti-4    	   50916	     21435 ns/op	    7638 B/op	      53 allocs/op
BenchmarkLogWithContext-4       	   54744	     20805 ns/op	   10675 B/op	      53 allocs/op
```

Reply:
@mfpierre am I reading this right? Did we have a 10x performance penalty after adding the lock?

Reply:
@truthbk sorry it was a bit misleading, adding the lock as minimal impact compared to the scrubbing

**No scrubbing:**
No lock:
`BenchmarkLogWithContext-4   	  111650	     10646 ns/op	    5818 B/op	      35 allocs/op`

Lock:
`BenchmarkLogWithContext-4   	  112789	     10056 ns/op	    5821 B/op	      35 allocs/op`

**Scrubbing:**
No lock:
`BenchmarkLogWithContext-4   	   67291	     19343 ns/op	   10761 B/op	      53 allocs/op`

Lock:
`BenchmarkLogWithContext-4   	   65120	     18912 ns/op	   10773 B/op	      53 allocs/op`

(I also pushed the commit adding the lock)